src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedElements = '';
+    const thisLength = this.length;
```

Comment:
Is this variable really necessary? You could just use `this.length` in the for loop.

Reply:
Yah I guess so, since I use it only once in the code. Thanx.